BackEndFlask/models/user_course.py

```diff
@@ -40,6 +40,13 @@ def get_user_course_by_user_id_and_course_id(user_id, course_id):
     except SQLAlchemyError as e:
         error = str(e.__dict__['orig'])
         return error
+
+def get_user_courses_by_user_id_and_role_id(user_id, role_id):
+    try:
+        return UserCourse.query.filter_by(user_id=user_id, role_id=role_id).first()
```

Comment:
I believe this should be .all() instead of .first() since it is trying to get all the user courses, not just the first one.

Reply:
Fixed!